src/riot-rs-threads/src/thread_flags.rs

```diff
@@ -1,9 +1,17 @@
 //! Thread flags.
+//!
+//! Every thread in RIOT-rs has a set of 16 flags that can be set and waited on.
+//!
+//! Note: The highest 4 bits (`1<<15, 1<<14, 1<<13, 1<<12`) are used inside
+//!       RIOT-rs and should not be used for applications.
 use crate::{ThreadId, ThreadState, Threads, THREADS};
 
 /// Bitmask that represent the flags that are set for a thread.
 pub type ThreadFlags = u16;
 
```

Comment:
We could enforce this through the API:

```suggestion
//!
//! Every thread in RIOT-rs has a set of 16 flags that can be set and waited on.
use crate::{ThreadId, ThreadState, Threads, THREADS};

/// Bitmask that represent the flags that are set for a thread.
pub struct ThreadFlags(u16);

/// Thread flag used to kick thread loops.
pub const THREAD_FLAG_WAKEUP: ThreadFlags = 1 << 15;

impl ThreadFlags {
    /// Create new [`ThreadFlags`].
    /// 
    /// The highest 4 bits (`1<<15, 1<<14, 1<<13, 1<<12`) are reserved
    /// for internal usage.
    /// If they are set in `flags`, `None` will be returned.
    pub fn new(flags: u16) -> Option<Self> {
        if flags >= 1 << 12 {
            return None;
        }
        return Some(ThreadFlags(flags))
    }

    /// Flag that signals a thread to wakup.
    pub(crate) fn wakeup_flag() -> Self {
        ThreadFlags(THREAD_FLAG_WAKEUP)
    }
}

```